internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/handler/ScanTask.java

```diff
@@ -609,38 +622,41 @@ private boolean uploadReportToFtp(String scanArtifact, String reportPath) {
         String scanReportFtpLocation = scanArtifact.substring(0, scanArtifact.lastIndexOf(File.separator));
 
         File reports = new File(reportPath);
-        try {
-            FileUtil.uploadReport(scanReportFtpLocation, reports,
-                    VeracodeScannerConfiguration.getInstance().getConfigProperty(ScannerConstants.FTP_USERNAME),
-                    (VeracodeScannerConfiguration.getInstance().getConfigProperty(ScannerConstants.FTP_PASSWORD))
-                            .toCharArray(), VeracodeScannerConfiguration.getInstance().getConfigProperty(
-                            ScannerConstants.FTP_HOST), Integer.parseInt(VeracodeScannerConfiguration.getInstance()
-                            .getConfigProperty(ScannerConstants.FTP_PORT)));
-
-            isReportUploaded = true;
-            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.COMPLETED,
-                    scanReportFtpLocation + File.separator + reports.getName(),
-                    null);
-
-            String logMessage = "Scan report is uploaded to the FTP server for the application: " +
-                    scanContext.getAppId();
-            log.info(logMessage);
-            CallbackUtil.persistScanLog(scanContext.getJobId(), logMessage, LogType.INFO);
-        } catch (SftpException | JSchException e) {
-            int retryInterval = Integer.parseInt(VeracodeScannerConfiguration.getInstance().getConfigProperty(
-                    VeracodeScannerConstants.SCAN_REPORT_UPLOAD_RETRY_SECONDS));
-            log.info("Report upload will retry after " + retryInterval + " seconds since that operation was failed " +
-                    "due to FTP server issue. \n" + ErrorProcessingUtil.getFullErrorMessage(e));
+
+        if (!Thread.currentThread().isInterrupted()) {
             try {
-                TimeUnit.SECONDS.sleep(retryInterval);
-            } catch (InterruptedException e1) {
-                log.error(e1);
+                FileUtil.uploadReport(scanReportFtpLocation, reports,
+                        VeracodeScannerConfiguration.getInstance().getConfigProperty(ScannerConstants.FTP_USERNAME),
+                        (VeracodeScannerConfiguration.getInstance().getConfigProperty(ScannerConstants.FTP_PASSWORD))
```

Comment:
Shall we assign these values to variables and then pass those in. It'll be more readable.